### DISCUSSION TITLE
Img2Img Alternative Test

### DISCUSSION THREAD

**Discussion by Nargodian** [Comfy-Org/ComfyUI#308] [Ideas]
So this feature (which was the secret sauce to Corridor digital's Anime Rock Paper Scissors) is not present as far as I can tell. 

The processes I think noises the image in such a way that the noise sort of maps to the image its self so that when you have similar images like frames of a video, the noise pattern is the same on similar objects. How it dose this I'm not 100% sure, but it would sure be useful to have.

Unless it does already exist which of course will a happy little mistake by me.

**Comment by WASasquatch** [comment]
I think to achieve this they're just Gaussian noise the tensor based on the mean and standard deviation of the pixel values. So each pixel value gets a relative effect, and similar pixels in similar images get the same effect. 

Something along the lines of this perhaps

```python
# Some image tensor
mean = torch.mean(image)
std = torch.std(image)
noise = torch.randn_like(image)
noisy_image = image + std * noise + mean
noisy_image = torch.clamp(noisy_image, 0, 1)
```

I have dyscalculia, and terrible with math, but what I envision here is the noise of a image sequence (video) would be more or less the same besides the actual moving bits of the frames.

**Reply by Nargodian** [reply]
So checking over the code in img2imgalt.py of A1111's ui, It seems to be a much more intricate process. But I'm no expert in this kind of stuff.

**Reply by WASasquatch** [reply]
> So checking over the code in img2imgalt.py of A1111's ui, It seems to be a much more intricate process. But I'm no expert in this kind of stuff.

Seems to be using de-noising from the model to compute it's noise, and then that result divided by standard deviation. 

Looks like it could be implemented. I am assuming the modes is for 2.x?

**Reply by Nargodian** [reply]
It works with 1.5, haven't tried it with any 2.x models
